src/resources/container.js

```diff
@@ -9,15 +9,35 @@ import { GlbParser } from './parser/glb-parser.js';
 /**
  * @class
  * @name pc.ContainerResource
- * @classdesc Container for a list of animations, textures, materials and a model.
+ * @classdesc Container for a list of animations, textures, materials, models, nodes, scenes, default scene
+ * cameras and lights. Entities in scene hierarchies will have model, camera and light components attached to them.
+ * Animation components have to be added manually (using nodeAnimations) as either pc.AnimComponent or pc.AnimationComponent.
  * @param {object} data - The loaded GLB data.
+ * @property {pc.Entity|null} scene Root entity of the default GLB scene.
+ * @property {pc.Entity[]} scenes Root entities of scenes indexed by GLB scenes.
+ * @property {pc.CameraComponent[]} cameras Instanced camera components, does not match index of GLB cameras.
+ * @property {pc.LightComponent[]} lights Instanced light components, does not match index of GLB lights.
+ * @property {pc.Entity[]} nodes Entities indexed by GLB nodes.
+ * @property {pc.Asset[]} materials Material assets indexed by GLB materials.
+ * @property {pc.Asset[]} textures Texture assets indexed by GLB textures.
+ * @property {pc.Asset[]} animations Animation assets indexed by GLB animations.
+ * @property {number[][]} nodeAnimations Animation asset indices indexed by GLB nodes.
+ * @property {pc.Asset[]} models Model assets indexed by GLB meshes.
+ * @property {pc.AssetRegistry} registry The asset registry.
  */
 function ContainerResource(data) {
     this.data = data;
-    this.model = null;
+    this.scene = null;
+    this.scenes = [];
+    this.cameras = [];
+    this.lights = [];
+    this.nodes = [];
     this.materials = [];
     this.textures = [];
     this.animations = [];
+    this.nodeAnimations = [];
+    this.models = [];
+    this._nodeModels = [];
```

Comment:
Is there maybe some more intuitive names for `nodeAnimations` and `_nodeModels `, that help indicate they are indices?

And why is `_nodeModels ` private compared to `nodeAnimations`?

Reply:
Good point.

 `_nodeModels` is private because it has no use outside of `ContainerResouce`. It is used in `ContainerResource` to apply model components to nodes and is kept by reference only so that the models can be properly destroyed upon unloading the container.

How about `nodeAnimationMapping`? Besides that  I can't think of a more intuitive name wouldn't be overly descriptive and long. If you think it would get rid of the confusion, I would actually prefer to rename `_nodeModels` instead. Something like `_appliedModels` perhaps. I don't feel that name has to be "as good" since it is not part of the public API. What do you think?

Reply:
Correct, `_nodeModels` contain models cloned from `models` per node. The reason models are cloned is to prevent properties applied for specific nodes from affecting the original model, which might be used by multiple nodes. Since the clones are not "related" to their originating models, using `model` indices won't work.

I don't think there is a way around the contents of `_nodeModels` (model ref per node) and `nodeAnimations` (anim indices per node) being different. For that reason I liked your second name suggestion:
* `_nodeModels` -> `_modelByNode`
* `nodeAnimations` -> `animationIndicesByNode` (I prefer `animation` over `anim` since `animation` is the term used in other parts of the ContainerResource API)

Should we go with this?

Reply:
Lets go with that.